api/src/routes/donate.ts

```diff
@@ -0,0 +1,61 @@
+import {
+  Type,
+  type FastifyPluginCallbackTypebox
+} from '@fastify/type-provider-typebox';
+
+export const donateRoutes: FastifyPluginCallbackTypebox = (
+  fastify,
+  _options,
+  done
+) => {
+  // The order matters here, since we want to reject invalid cross site requests
+  // before checking if the user is authenticated.
+  // eslint-disable-next-line @typescript-eslint/unbound-method
+  fastify.addHook('onRequest', fastify.csrfProtection);
+  fastify.addHook('onRequest', fastify.authenticateSession);
+
+  fastify.post(
+    '/add-donation',
+    {
+      schema: {
+        body: Type.Object({
+          isDonating: Type.Boolean()
+        }),
+        response: {
+          200: Type.Object({
+            message: Type.String(),
+            type: Type.Literal('success')
+          }),
+          500: Type.Object({
+            message: Type.Literal('Something went wrong.'),
+            type: Type.Literal('danger')
+          })
+        }
+      }
+    },
+    async (req, reply) => {
+      try {
+        await fastify.prisma.user.update({
+          where: { id: req.session.user.id },
+          data: {
+            isDonating: req.body.isDonating
+          }
+        });
+
+        return {
+          message: 'success',
+          type: 'success'
+        } as const;
```

Comment:
Do we want to send a more detailed message?

(looks like the current API just sends back the `{ isDonating: true }` object)

Reply:
Ideally it should clone the existing api unless we're 100% sure the changes don't matter.